### PR TITLE
Change data loading pattern on front page, configure Tailwind

### PR DESCRIPTION
**Pull request by MatthewCaseres**
## Preview this change

See my changes at https://www.omsreviews.tech/. Refresh the page a couple of times and compare it with https://omshub.org/ to see that it is faster because it is built statically.

## Contents of PR

Data is updated automatically using [Incremental Static Regeneration (ISR)](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration) with a refresh rate of 1 minute. This allows the site traffic to not cause any additional stress on the server even during peak traffic, because pages are statically built.

Tailwind styling is not used in this refactoring, but I want to work with them in the future for certain custom components and layouts.

### CODE REVIEW

File: .devcontainer/scripts/start.sh [hunk 1]
```diff
@@ -3,5 +3,4 @@
 # Performs setup tasks _after_ the .devcontainer Docker container is created.
 
 # Install Node dependencies.
-yarn install
-yarn build
```
Comment (awpala) [inline .devcontainer/scripts/start.sh:7]: We should leave this here for now, it only impacts the Docker container here FYI, but is consistent with the README instructions pertaining to Docker

Comment (awpala) [inline .devcontainer/scripts/start.sh:7]: We should leave this here for now, it only impacts the Docker container here FYI, but is consistent with the README instructions pertaining to Docker

Comment (MatthewCaseres) [reply .devcontainer/scripts/start.sh:7]: Changed the README then cause instructions in README weren't quite right.

**Comment by MatthewCaseres** [comment]
I'm gonna reopen this on a clean branch

**Pull request closed without merging**